AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,115 @@
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UINavigationControllerDelegate, UIImagePickerControllerDelegate {
     var titleForCell: String = "Double Video"
-
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
+    var imagePickerController = UIImagePickerController()
+    var videoURL: URL?
+    var topPlayer: AVPlayer!
+    var bottomPlayer: AVPlayer?
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        //self.videoContainerBottom.backgroundColor = .red
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
+        
     }
 
     override func viewDidLayoutSubviews() {
-        // update your layers' frames here
+//        guard let sublayersTop = self.videoContainerTop.layer.sublayers
+//            else {
+//                return
+//        }
+//        for layer in sublayersTop {
+//            layer.frame = self.videoContainerTop.bounds
+//        }
+// 
+//        guard let sublayersbottom = self.videoContainerBottom.layer.sublayers
+//            else {
+//                return
+//        }
+//        for layer in sublayersbottom {
+//            layer.frame = self.videoContainerBottom.bounds
+//        }
+        let topContainer = videoContainerTop.layer.sublayers
+        let bottomContainer = videoContainerBottom.layer.sublayers
+        
+        if topContainer != nil && bottomContainer != nil {
+            topContainer?[0].frame = self.videoContainerTop.bounds
+        }
+        
+        if topContainer != nil && bottomContainer == nil {
+            topContainer?[0].frame = self.videoContainerTop.bounds
+        } else {
+            bottomContainer?[0].frame = self.videoContainerBottom.bounds
+        }
+        
     }
-    
+
     @IBAction func loadVideo(_ sender: UIButton) {
+        imagePickerController.sourceType = .photoLibrary
+        imagePickerController.mediaTypes = [String(kUTTypeMovie)]
+        imagePickerController.delegate = self
+        self.present(imagePickerController, animated: true, completion: nil)
+    }
     
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+            self.videoURL = url
+        }
+        self.dismiss(animated: true) {
+//            if self.topPlayer.rate != 0.0 && self.bottomPlayer.rate != 0.0 {
+//                let alert = UIAlertController(title: "Error", message: "Videos are playing on both", preferredStyle: .alert)
+//                let ok = UIAlertAction(title: "OK", style: .cancel, handler: nil)
+//                alert.addAction(ok)
+//                self.present(alert, animated: true, completion: nil)
+//
+//            }
+            switch self.videoContainerTop.layer.sublayers?.count {
```

Comment:
I think it was a good intuition to use a switch for a condition which might have more than a couple of states but because sublayers is optional it turns count into an optional when chained and things like ```case 1?:``` result that are kind of hard to read or understand. Also, in testing I see that the video is cleared out (bonus points awarded) but I never see two videos play at once.